ckanext/validation/jobs.py

```diff
@@ -395,3 +410,68 @@ def _prep_foreign_keys(package, table_schema, resource, df):
         for field in table_schema['fields']:
             if field['name'] in foreign_keys.keys():
                 field['foreignKey'] = foreign_keys[field['name']]
+
+
+def _reorder_columns(schema, df):
+    """
+    Creating a preset in which you can configure whether the order of columns matters.
+    """
+
+    log.debug(df.dtypes)
+    log.debug("Considering whether to switch column order")
+
+    required_field_order = list(map(
+        lambda x: x['name'],
+        schema.get('fields', [])
+    ))
+    submitted_field_order = list(df.columns)
+    errors = {}
+    for f in required_field_order:
+        if f in submitted_field_order:
+            submitted_field_order.remove(f)
```

Comment:
We need to finish with a list that matches submitted field order in everyway (including element order) except it has had the required_field_order  elements removed from it.  We also need to manage errors.  So another way of doing it would be:

Reply:
Ah I've just seen your next comment.